src/daemon/daemon.cpp

```diff
@@ -760,18 +760,13 @@ std::unordered_set<std::string> mac_set_from(const mp::VMSpecs& spec)
     return macs;
 }
 
-// Move the contents of t to s. If both sets are disjoint, then s becomes sUt, t becomes valid but unspecified (for
-// efficiency reasons) and true is returned. If they are not disjoint, then s and t remain untouched and false is
-// returned.
-bool merge_if_disjoint(std::unordered_set<std::string>& s, std::unordered_set<std::string>& t)
+// Merge the contents of t into s, iff the sets are disjoint (i.e. make s = sUt). Return whether s and t were disjoint.
+bool merge_if_disjoint(std::unordered_set<std::string>& s, const std::unordered_set<std::string>& t)
 {
-    for (const auto& elt : s)
-        if (t.find(elt) != t.end())
-            return false;
-
-    for (auto& elt : t)
-        s.insert(std::move(elt));
+    if (any_of(cbegin(s), cend(s), [&t](const auto& mac) { return t.find(mac) != cend(t); }))
+        return false;
 
+    s.insert(cbegin(t), cend(t));
     return true;
 }
```

Comment:
The thing is, this actually fixes a bug that I hadn't noticed before!

Consider these two commits: [A](https://github.com/canonical/multipass/pull/1875/commits/44652c51e62ba4e847613938312c8d77be6da1da) and [B](https://github.com/canonical/multipass/pull/1875/commits/2c192bdaca4f1681ad9ca74500343ab2448622dc). Without A, any instances that failed to create would be discarded, but still have their macs saved. But once we have A, we need B, because otherwise we would end up with macs that were moved-from.

Reply:
Right, I didn't realize that! Fine then.